e2e/test/prerequisites/E2ETestsSetup/e2eTestsSetup.ps1

```diff
@@ -80,6 +86,22 @@ Function Connect-AzureSubscription()
     return $azureContext
 }
 
+Function Check-AzureCliVersion()
+{
+    $azCliVersionTested = [System.Version]"2.37.0"
+
+    $azCliVersionCurrentString = az version --query '\"azure-cli\"'
+    $azCliVersionCurrent = [System.Version]($azCliVersionCurrentString.Trim('"'))
+
+    if ($azCliVersionTested -gt $azCliVersionCurrent)
+    {
+        Write-Host "`nVersion of Azure CLI installed is $azCliVersionCurrent while this script has been tested on a newer version of $azCliVersionTested."
+        Write-Host "`nUpdating Azure CLI version to $azCliVersionTested."
+
+        az upgrade
```

Comment:
az cli doesn't support version specific upgrade command, i.e. we cannot do `az upgrade --v <target-version-required>`.
So, this command will always upgrade to latest.